### ISSUE TITLE
Not receiving messages on Desktop with KDE Connect 1.12.9

### ISSUE THREAD

**Issue by AlicesReflexion** [andyholmes/gnome-shell-extension-gsconnect#581]
**Describe the bug**

Chat history is no longer synced with KDE Connect with versions newer than 1.12.7.

Upon upgrade to 1.12.9, messages stop syncing. Downgrade, and messages start syncing again. 

Message notifications do show up. Sending a message works, but the desktop never registers the message as sent, and just leaves it hanging indefinitely. 

Deleting the SMS history, then re-importing from a backup, as in #456 does not fix the issue. May be related to #541.

**Steps To Reproduce:**

Open GSConnect, go to messaging. Send a text, it never registers as sent. Receive a text, it never shows up at all.

**Expected behavior**

GSConnect Messaging updates as soon as a text is received.

**Screenshots**

After clearing SMS cache, everything is completely gone.
![conversation](https://user-images.githubusercontent.com/6933425/62741841-fa627180-ba01-11e9-8779-b9f8658532ba.jpg)

**Support Log**

Nothing useful as far as I can tell.
I will provide an ADB log when I get my hands on a USB-C <-> USB-C cable, either later today or early tomorrow.

```

GSConnect Version: 24
GSConnect Install: user
GJS: 15602
XDG_SESSION_TYPE: wayland
GDMSESSION: gnome
--------------------------------------------------------------------------------
-- Logs begin at Thu 2019-08-08 10:43:57 CDT, end at Thu 2019-08-08 17:13:33 CDT. --
 8月 08 17:12:02 gnome-shell[18074]: [AppIndicatorSupport-WARN] Attempting to re-register :1.329/org/ayatana/NotificationItem/duplicati; resetting instead
 8月 08 17:12:02 gnome-shell[18074]: [AppIndicatorSupport-WARN] Item :1.329/org/ayatana/NotificationItem/duplicati is already registered
 8月 08 17:12:02 gnome-shell[18074]: [AppIndicatorSupport-WARN] Attempting to re-register :1.329/org/ayatana/NotificationItem/duplicati; resetting instead
 8月 08 17:12:02 gnome-shell[18074]: [AppIndicatorSupport-WARN] Item :1.329/org/ayatana/NotificationItem/duplicati is already registered
 8月 08 17:12:29 systemd[2034]: dbus-:1.2-org.gnome.gedit@2.service: Succeeded.
 8月 08 17:13:08 org.gnome.Shell.Extensions.GSConnect[18578]: [/service/plugins/notification.js:sendNotification:352]: "(Firefox) Sailor Haumea 🌐: @SailorHaumeaさんにフォローされました"
 8月 08 17:13:08 org.gnome.Shell.Extensions.GSConnect[18578]: [/service/protocol/core.js:send:220]: Sony: {
                                                                 "id": 1565302388615,
                                                                 "type": "kdeconnect.notification",
                                                                 "body": {
                                                                   "appName": "Firefox",
                                                                   "id": "fdo|null|0",
                                                                   "title": "Sailor Haumea 🌐",
                                                                   "text": "@SailorHaumeaさんにフォローされました",
                                                                   "ticker": "Sailor Haumea 🌐: @SailorHaumeaさんにフォローされました",
                                                                   "isClearable": false
                                                                 }
                                                               }
 8月 08 17:13:08 NetworkManager[1393]: <info>  [1565302388.6395] policy: set 'TryFi-5Ghz 1' (wlp2s0) as default for IPv6 routing and DNS
 8月 08 17:13:09 goa-daemon[2432]: Remote error from secret service: org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
 8月 08 17:13:09 goa-daemon[2432]: secret_password_lookup_sync() failed: The name is not activatable
 8月 08 17:13:10 goa-daemon[2432]: Remote error from secret service: org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
 8月 08 17:13:10 goa-daemon[2432]: secret_password_lookup_sync() failed: The name is not activatable
 8月 08 17:13:16 gnome-shell[18074]: value "-1.000000" of type 'gfloat' is invalid or out of range for property 'width' of type 'gfloat'
 8月 08 17:13:18 org.gnome.Shell.Extensions.GSConnect[18578]: [/service/protocol/core.js:send:220]: Sony: {
                                                                 "id": 1565302398990,
                                                                 "type": "kdeconnect.sms.request",
                                                                 "body": {
                                                                   "sendSms": true,
                                                                   "phoneNumber": "******",
                                                                   "messageBody": "wait what"
                                                                 }
                                                               }
 8月 08 17:13:20 goa-daemon[2432]: Remote error from secret service: org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
 8月 08 17:13:20 goa-daemon[2432]: secret_password_lookup_sync() failed: The name is not activatable
 8月 08 17:13:21 goa-daemon[2432]: Remote error from secret service: org.freedesktop.DBus.Error.ServiceUnknown: The name is not activatable
 8月 08 17:13:21 goa-daemon[2432]: secret_password_lookup_sync() failed: The name is not activatable
 8月 08 17:13:33 org.gnome.Shell.Extensions.GSConnect[18578]: [/service/protocol/core.js:receive/<:177]: Sony: {
                                                                 "id": 1565302412831,
                                                                 "type": "kdeconnect.notification",
                                                                 "body": {
                                                                   "payloadHash": "8fe1276fb5e9dcec1cc7fd0277189b7e",
                                                                   "actions": [
                                                                     "既読にする"
                                                                   ],
                                                                   "id": "0|com.google.android.apps.messaging|0|com.google.android.apps.messaging:sms:78|10120",
                                                                   "isClearable": true,
                                                                   "appName": "メッセージ",
                                                                   "time": "1565302412705",
                                                                   "requestReplyId": "2ddc56c6-8a46-4597-a3df-a7433505c2b1",
                                                                   "ticker": "****: ? ",
                                                                   "title": "****",
                                                                   "text": "? \n"
                                                                 },
                                                                 "payloadSize": 1651,
                                                                 "payloadTransferInfo": {
                                                                   "port": 1739
                                                                 }
                                                               }
```

**System Details (please complete the following information):**

 - **GSConnect version:** 25-rc1, but also broken on 24.
   - **Installed from:** Github
 - **GNOME/Shell version:** 3.32.2
 - **Distro/Release:** Fedora 30

**GSConnect environment (if applicable):**

 - **Paired Device(s):** Xperia XA2
 - **KDE Connect app version:** 1.12.93
 - **Plugin(s):** SMS sync

**Additional Notes:**

Language on phone and desktop is Japanese, but I do not see this making a difference.

**Comment by ferdnyc** [comment]
@AlicesReflexion 

If SMS syncing works with KDE Connect 1.12.7 and _doesn't_ with 1.12.9, that sounds very much like an issue for the KDE Connect app developers. Please do search for / create a bug for [kdeconnect in kde bugzilla](https://bugs.kde.org/buglist.cgi?quicksearch=kdeconnect&list_id=1505828) with details about the issue.

I'm going to leave this open for the moment, though, because of the "Send a text, it never registers as sent. Receive a text, it never shows up at all" component. That may indicate that something has changed in the communication between GSConnect and KDE Connect with the 1.12.9 release which may need to be addressed on the GSConnect end. @andyholmes ?

**Comment by Huluti** [comment]
Also affetected by this bug with GS Connect 24.
![image](https://user-images.githubusercontent.com/5782198/63226281-639f6d00-c1d8-11e9-84d2-19bb0c058ec7.png)

**Comment by andyholmes** [comment]
Sorry I missed this one earlier. Assuming everyone experiencing this bug has a log like the one posted by @AlicesReflexion, there doesn't seem to be a problem in GSConnect. In that case, this probably is a bug kdeconnect-android, or something has changed for only some phones that I'm not aware of.

For historical reasons, the <kbd>+</kbd> button will send a request for the SMS history and Contacts every time it's pressed. If you can get `adb` setup with your phone, you can run this command to list the Anroid app's debug output:

```sh
adb logcat --pid=$(adb shell pidof -s org.kde.kdeconnect_tp)
```

Then see if there are any errors or useful output when you press the <kbd>+</kbd> button. If there are you should open a bug on the KDE bug tracker and possible ping `@sredman` on this issue, since we usually CC each other on SMS bugs that affect both projects.

**Comment by ferdnyc** [comment]
Since I've _never_ gotten conversation history from my phone (a stock-ROM Samsung Galaxy S6 from Sprint) — but I do get contacts — and I have a shiny new USB cable that actually works to make an `adb` connection without flaking out like my previous one did, I decided to try this just for fun. The results are probably typical and sadly uninformative, pretty much as expected.

```
09-04 22:26:15.757 16755 23834 E KDE/Device: Exception in SMSPlugin's onPacketReceived()
09-04 22:26:15.757 16755 23834 E KDE/Device: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at android.os.Parcel.readException(Parcel.java:1699)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at android.content.ContentProviderProxy.query(ContentProviderNative.java:421)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at android.content.ContentResolver.query(ContentResolver.java:536)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at android.content.ContentResolver.query(ContentResolver.java:478)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Helpers.SMSHelper.getMessages(SMSHelper.java:215)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Helpers.SMSHelper.getConversations(SMSHelper.java:308)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Plugins.SMSPlugin.SMSPlugin.handleRequestConversations(SMSPlugin.java:365)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Plugins.SMSPlugin.SMSPlugin.onPacketReceived(SMSPlugin.java:301)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Device.onPacketReceived(Device.java:553)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Backends.BaseLink.packageReceived(BaseLink.java:82)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.receivedNetworkPacket(LanLink.java:252)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.lambda$reset$0$LanLink(LanLink.java:107)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.-$$Lambda$LanLink$XXXREDACTED_JUST_IN_CASEXXX.run(lambda)
09-04 22:26:15.757 16755 23834 E KDE/Device: 	at java.lang.Thread.run(Thread.java:762)
```

**Comment by AlicesReflexion** [comment]
This is fixed with KDE Connect 1.13 and GSConnect 25-rc3
